### PR TITLE
make sure to copy attributes on subtraction

### PR DESCRIPTION
**Pull request by dannyjacobs**
Subtracting two datacontainers returned a new datacontainer shorn of all attributes. This fixes the problem by always returning the difference with all the attributes of the A, where the difference being computed is A - B.

### CODE REVIEW

File: hera_cal/datacontainer.py [hunk 1]
```diff
@@ -243,22 +243,21 @@ def __sub__(self, D):
...
             # iterate over D keys
             for i, k in enumerate(D.keys()):
                 if self.__contains__(k):
                     newD[k] = self.__getitem__(k) - D[k]
 
-            return DataContainer(newD)
+            return newD
 
         else:
             newD = copy.deepcopy(self)
```
Comment (jsdillon) [inline hera_cal/datacontainer.py:263]: This line is now redundant.

File: hera_cal/datacontainer.py [hunk 2]
```diff
@@ -243,22 +243,21 @@ def __sub__(self, D):
         otherwise subtract D from each ndarray in self.
```
Comment (jsdillon) [inline hera_cal/datacontainer.py:243]: I'd edit the docstring to clarify that the original object is always duplicated and that the metadata of the original object is preserved (but not the new object).

**Review by jsdillon — CHANGES_REQUESTED** [review]
Thanks @dannyjacobs!

If you're going to make this change, I'd suggest making parallel changes to `__add__` and `__mult__` and `__floordiv__` and `__truediv__`. I'd also like to see at least one unit test that exposes the issue you were having and is fixed by this change.

**Comment by dannyjacobs** [comment]
Have made the following changes:
1. Change propagated to add, sub, mul, div
2. Docstring informs about returning a copy object
3. updated test so that it fails without this fix.
4. branch adder_fail adds just the test to the main branch. See main fail!

**Review by jsdillon — APPROVED** [review]
Assuming my change passes tests, I'm good to go. Thanks @dannyjacobs !

**Pull request merged**